pkg/endpoint/bpf.go

```diff
@@ -478,6 +473,11 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir, reason string) (uint64, err
 			e.Mutex.Unlock()
 			return 0, fmt.Errorf("Unable to regenerate policy: %s", err)
 		}
+
+		if err = e.writeHeaderfile(epdir, owner); err != nil {
```

Comment:
I think it'd be helpful to add a comment illustrating why we invoke this after regenerating policy within the code itself.

Reply:
I'll open an issue to restructure this. This is a quickfix to unblock #2708. There are several critical bugfixes here including adding the necessary comments.

Reply:
Works for me, thanks!